cloudinit/config/schema.py

```diff
@@ -66,23 +65,25 @@
 DEPRECATED_KEY = "deprecated"
 
 
-# annotations add value for development, but don't break old versions
-# pyver: 3.6 -> 3.8
-# pylint: disable=E1101
-if sys.version_info >= (3, 8):
+# type-annotate only if type-checking.
+# Consider to add `type_extensions` as a dependency when Bionic is EOL.
+if TYPE_CHECKING:
```

Comment:
> Explicit runtime checks for the win. Thanks for this to avoid trying to import typing modules when we happen to be in runtime.

Yeah!


> One thing I don't think we can do though is drop the additional check for 3,8 without some level of checking that import TypedDict is present. As it is, python 3.6 and 3.7 are still 'supported', so doesn't that leave us with exposure to typing.TypedDict not existing yet?

Maybe it went unnoticed, but we are importing `TypeDict` from `typing_extensions` which provides a back-ported implementation of `TypeDict` for 3.6 and 3.7. Therefore, we do not need additional handling.